### ISSUE TITLE
Prepare nested tibble for writing to database

### ISSUE THREAD

**Issue by krlmlr** [cynkra/dm#732]
- [ ] Convert nested columns to NDJSON
- [ ] Flatten packed columns using `"__"` as separator

``` r
library(tidyverse)
library(dm)

x <- dm::dm_wrap(dm_nycflights13(), airlines)

to_ndjson <- function(x) {
  con <- textConnection(NULL, open = "w")
  on.exit(close(con))
  jsonlite::stream_out(x, con, verbose = FALSE)
  paste0(textConnectionValue(con), "\n", collapse = "")
}

arrow <- x$airlines
arrow$flights <- map_chr(arrow$flights, to_ndjson)

roundtrip_arrow <- arrow

roundtrip_arrow$flights <- vctrs::as_list_of(map(roundtrip_arrow$flights, ~ arrow::read_json_arrow(charToRaw(.x))))

roundtrip_arrow
#> # A tibble: 15 × 3
#>    carrier name                                    flights
#>    <chr>   <chr>                       <list<tibble[,21]>>
#>  1 9E      Endeavor Air Inc.                    [104 × 21]
#>  2 AA      American Airlines Inc.               [180 × 21]
#>  3 AS      Alaska Airlines Inc.                   [4 × 21]
#>  4 B6      JetBlue Airways                      [296 × 21]
#>  5 DL      Delta Air Lines Inc.                 [239 × 21]
#>  6 EV      ExpressJet Airlines Inc.             [278 × 21]
#>  7 F9      Frontier Airlines Inc.                 [4 × 21]
#>  8 FL      AirTran Airways Corporation           [20 × 21]
#>  9 HA      Hawaiian Airlines Inc.                 [2 × 21]
#> 10 MQ      Envoy Air                            [147 × 21]
#> 11 UA      United Air Lines Inc.                [294 × 21]
#> 12 US      US Airways Inc.                      [109 × 21]
#> 13 VX      Virgin America                        [19 × 21]
#> 14 WN      Southwest Airlines Co.                [62 × 21]
#> 15 YV      Mesa Airlines Inc.                     [3 × 21]
```

<sup>Created on 2022-01-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by krlmlr** [comment]
JSON is better, because it's consistent with what (at least) SQL Server does with `FOR JSON` .

**Comment by krlmlr** [comment]
Part of #913.

**Comment by krlmlr** [comment]
Solved by #913.
